src/components/auth/EmailResetPassword.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import { useEffect, useInsertionEffect, useMemo, useState } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { useResetPasswordMutation } from "@/hooks/mutations/reset-password";
+import { useEmail } from "@/hooks/query/email";
+import { TextInput } from "../TextInput";
+import { Button } from "../ui/button";
+import { useToast } from "../ui/use-toast";
+import { Card } from "../Card";
+import { SkeletonEmailResetPassword } from "./SkeletonEmailResetPassword";
+
+const formSchema = z.object({
```

Comment:
This is used in the sign up form as well right? Can we find a common place to put this schema and export it?